src/NodeApi/JSValue.cs

```diff
@@ -21,38 +21,87 @@ namespace Microsoft.JavaScript.NodeApi;
 
     internal JSRuntime Runtime => Scope.Runtime;
 
+    /// <summary>
+    /// Creates an empty instance of <see cref="JSValue" />, which implicitly converts to
+    /// <see cref="JSValue.Undefined" /> when used in any scope.
+    /// </summary>
     public JSValue() { }
 
-    public JSValue(napi_value handle) : this(handle, JSValueScope.Current)
-    {
-    }
+    /// <summary>
+    /// Creates a new instance of <see cref="JSValue" /> from a handle in the current scope.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: A JS value handle is a pointer to a location in memory, so an invalid handle here
+    /// may cause an attempt to access an invalid memory location.
+    /// </remarks>
+    public JSValue(napi_value handle) : this(handle, JSValueScope.Current) { }
 
+    /// <summary>
+    /// Creates a new instance of <see cref="JSValue" /> from a handle in the specified scope.
+    /// </summary>
+    /// <exception cref="ArgumentNullException">Thrown when the scope is null (unless the handle
+    /// is also null).</exception>
+    /// <remarks>
+    /// WARNING: A JS value handle is a pointer to a location in memory, so an invalid handle here
+    /// may cause an attempt to access an invalid memory location.
+    /// </remarks>
     public JSValue(napi_value handle, JSValueScope? scope)
     {
         if (!handle.IsNull && scope is null) throw new ArgumentNullException(nameof(scope));
         _handle = handle;
         _scope = scope;
     }
 
-    public napi_value? Handle
-        => !Scope.IsDisposed ? (_handle.Handle != default(nint) ? _handle : Undefined._handle) : null;
+    /// <summary>
+    /// Gets the value handle, or throws an exception if the value scope is disposed or
+    /// access from the current thread is invalid.
+    /// </summary>
+    /// <exception cref="JSValueScopeClosedException">The scope has been closed.</exception>
+    /// <exception cref="JSInvalidThreadAccessException">The scope is not valid on the current
+    /// thread.</exception>
+    public napi_value Handle
+    {
+        get
+        {
+            if (_scope == null)
+            {
+                // If the scope is null, this is an empty (uninitialized) instance.
+                // Implicitly convert to the JS `undefined` value.
+                return Undefined._handle;
+            }
+
+            // Ensure the scope is valid and on the current thread (environment).
+            _scope.ThrowIfDisposed();
+            _scope.ThrowIfInvalidThreadAccess();
+
+            // The handle must be non-null when the scope is non-null.
```

Comment:
>// The handle must be non-null when the scope is non-null. [](http://example.com/codeflow?start=12&length=58)

Should we add a check in the constructor for it?
Currently we only force scope to be not null if handle is not empty. We never check that handle is not null when handle is not null.
Previously we used to check for null handle: `Handle ?? throw new ...`

Reply:
Done.